js/util/misc.ts

```diff
@@ -27,7 +27,9 @@ export const compareStudentsByName = (student1: Map<string, any>, student2: Map<
 export const answerHash = (answer: Map<string, any>) => {
   let answerContent = answer.get("answer");
   if (typeof answerContent !== "string") {
-    answerContent = JSON.stringify(answerContent.toJS());
+    answerContent = Map.isMap(answerContent)
+      ? JSON.stringify(answerContent.sortBy((v: any, k: any) => k).toJS())
+      : JSON.stringify(answerContent.toJS());
```

Comment:
@mklewandowski can you add a test for this code?

I'd guess that this has been a problem for a long time now so feedback on image questions and probably generic interactives has had this problem for a while. We could try to do something fancy, to fix the old data, but I don't think it is worth it.

Reply:
FYI, the npm library I mentioned in standup that handles this is [object-hash](https://www.npmjs.com/package/object-hash). In this case, you could use `answerContent = object-hash(answerContent.toJS())`, although that wouldn't be backwards-compatible with any existing hashes.

Reply:
@scytacki I added unit tests to test both the previous working case and this newly working case where the answer itself is a Map.